cli/mender-artifact/dump.go

```diff
@@ -0,0 +1,242 @@
+// Copyright 2019 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"strings"
+
+	"github.com/mendersoftware/mender-artifact/areader"
+	"github.com/mendersoftware/mender-artifact/artifact"
+	"github.com/mendersoftware/mender-artifact/handlers"
+
+	"github.com/pkg/errors"
+	"github.com/urfave/cli"
+)
+
+type artifactDump struct {
```

Comment:
I don't really see the need for algebraics here. a simple `type cleverVarName []string` should be good enough.

Reply:
Haha, indeed. I originally expected this struct to contain more fields. Overengineering at its best! I'll change it.